java/videostore/src/Customer.java

```diff
@@ -1,67 +1,81 @@
-			switch (each.getMovie ().getPriceCode ()) {
-				case Movie.REGULAR:
-					thisAmount += 2;
-					if (each.getDaysRented () > 2)
-						thisAmount += (each.getDaysRented () - 2) * 1.5;
-					break;
-				case Movie.NEW_RELEASE:
-					thisAmount += each.getDaysRented () * 3;
-					break;
-				case Movie.CHILDRENS:
-					thisAmount += 1.5;
-					if (each.getDaysRented () > 3)
-						thisAmount += (each.getDaysRented () - 3) * 1.5;
-					break;
-			}
-
-			frequentRenterPoints++;
-
-			if (each.getMovie ().getPriceCode () == Movie.NEW_RELEASE
-					&& each.getDaysRented () > 1)
-				frequentRenterPoints++;
-
-			result += "\t" + each.getMovie ().getTitle () + "\t"
-								+ String.valueOf (thisAmount) + "\n";
-			totalAmount += thisAmount;
-
-		}
-
-		result += "You owed " + String.valueOf (totalAmount) + "\n";
-		result += "You earned " + String.valueOf (frequentRenterPoints) + " frequent renter points\n";
-
-
-		return result;
-	}
-
-
-	private String name;
-	private Vector rentals = new Vector ();
+import java.util.ArrayList;
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private List<Rental> rentals = new ArrayList<>();
+    private double amountOwed = 0;
+    private int frequentRenterPoints = 0;
+    double currentAmount = 0;
+
+    public Customer(String name) {
+        this.name = name;
+    }
+
+    public void addRental(Rental rental) {
+        rentals.add(rental);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String statement() {
+        String result = "Rental Record for " + getName() + "\n";
+
+        result = updateAmountOwedAndRenterPoints(result);
+
+        result += "You owed " + String.valueOf(amountOwed) + "\n";
+        result += "You earned " + String.valueOf(frequentRenterPoints) + " frequent renter points\n";
+
+
+        return result;
+    }
+
+    public double getAmountOwed() {
+        return amountOwed;
+    }
+
+    public int getFrequentRenterPoints() {
+        return frequentRenterPoints;
+    }
+
+    private String updateAmountOwedAndRenterPoints(String result) {
+        for (Rental eachMovie : rentals) {
+            amountOwed += getAmountAccruedForRentals(currentAmount, eachMovie);
+
+            getFrequentRenterPoints(eachMovie);
+
+            result += "\t" + eachMovie.getMovie().getTitle() + "\t" + String.valueOf(currentAmount) + "\n";
+        }
+        return result;
+    }
+
+    private int getFrequentRenterPoints(Rental eachMovie) {
+        if (eachMovie.getMovie().getPriceCode() == Movie.NEW_RELEASE && eachMovie.getDaysRented() > 1) {
+            frequentRenterPoints += 2;
+        } else {
+            frequentRenterPoints++;
+        }
+        return frequentRenterPoints;
+    }
+
+    private double getAmountAccruedForRentals(double currentAmount, Rental each) {
+        switch (each.getMovie().getPriceCode()) {
+            case Movie.REGULAR:
+                currentAmount += 2;

```

Comment:
This is also a dependency on a global/external variable.
This method should only return the points that this single Rental awards.